### DISCUSSION TITLE
アップロードしたファイルについて改善要望

### DISCUSSION THREAD

**Discussion by Brutus03** [increments/qiita-discussions#516] [要望・アイデア・不具合]
ご担当者　様

いつからか、設定画面より「アップロードしたファイル」の管理ができるようになったと思います。

この機能について、投稿した記事と紐づいているか分かると、良いと思います。
理由として、意図せずアップロードしてしまったファイルを後から確認する際に、発見しやすいです。
また、Qiitaとしても画像の保存に関して、awsのs3を利用しているかと思いますが、不要なファイルをユーザー側で削除しやすくなると、結果S3に関するコストの削減にも繋がると思います。

以上です。よろしくお願いします。

**Comment by tanaka-asami415** [comment]
@Brutus03 
ご意見ありがとうございます！

こちらのリリースでアップロードしたファイルを削除できるようになり
https://qiita.com/release-notes/2023-01-17

こちらのリリースでアップロードしたファイルのURLをコピーできるようにしました。
[https://qiita.com/release-notes/2023-05-10](https://qiita.com/release-notes/2023-05-10#%E3%82%A2%E3%83%83%E3%83%97%E3%83%AD%E3%83%BC%E3%83%89%E3%81%97%E3%81%9F%E3%83%95%E3%82%A1%E3%82%A4%E3%83%AB%E3%81%AEurl%E3%82%92%E3%82%B3%E3%83%94%E3%83%BC%E3%81%99%E3%82%8B%E6%A9%9F%E8%83%BD%E3%82%92%E5%85%AC%E9%96%8B%E3%81%97%E3%81%BE%E3%81%97%E3%81%9F)

> この機能について、投稿した記事と紐づいているか分かると、良いと思います。

今回いただいた機能を実現するには、現在Qiitaでは「画像がどの記事に紐づいているか」という情報を保存していないため
ユーザーが投稿した記事やコメント、編集リクエスト等すべてのコンテンツの中から各画像URLが含まれているか探す必要があります。

このような理由から、すぐに実装することは困難な状況です。

ゆくゆくは検討したいと思いますが、直近の開発優先度などから考えるとかなり先となりそうです🙏
ご不便をおかけして申し訳ないですが、ご了承いただけますと幸いです🙇

**Comment by Brutus03** [comment]
> ゆくゆくは検討したいと思いますが、直近の開発優先度などから考えるとかなり先となりそうです🙏

今のところ私としては、ただ管理コストが増えたしか効果を感じないため、改善のほどお願いします。

以上です。よろしくお願いします。
